### PR TITLE
Improve keyboard accessibility for review flow and other forms

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/6579 and fixes https://github.com/mozilla/addons-frontend/issues/6246 by using buttons instead of anchors.

There is some code lore about how we switched from buttons to links in order to workaround a bug in Firefox for Android that triggered hover styles on the first tap (see https://github.com/mozilla/addons-frontend/issues/3458). It would require two taps to trigger the button but the problem went away with an anchor tag. That original problem affected Firefox 58. I tested today with Firefox 62 (release) on Android and I no longer see the problem. I'm not sure when it was fixed but I think it's safe to proceed here.

This patch fixes the same problem for all instances of `<DismissibleText>`. I have manually tested those for regressions.

### CODE REVIEW

File: src/ui/components/DismissibleTextForm/index.js [hunk 1]
```diff
@@ -154,55 +154,48 @@ export class DismissibleTextFormBase extends React.Component<
           <div className="DismissibleTextForm-formFooter">{formFooter}</div>
         )}
         <div className="DismissibleTextForm-buttons">
-          {/*
-            These buttons each have an href so that they become anchor tags.
-            This prevents mobile taps from triggering their hover styles.
-          */}
-          {onDismiss && (
+          <span className="DismissibleTextForm-delete-submit-buttons">
```
Comment (kumar303) [inline src/ui/components/DismissibleTextForm/index.js]: This diff is confusing. I just switched the order of the buttons to make them keyboard accessible. When you press tab, the submit button is now first.

**Comment by muffinresearch** [comment]
I'm wondering about the specific rationale behind having source order one way and changing the visual order in CSS. I'm thinking that it's usually desirable  for the tabbing order to follow the visual layout ordering and to not override that, but it's not clear if part of the issue here is to do with one order for the column vs row layout.

Another consideration is that by default hitting the `enter` key should work for form submission (the submit button would need `type="submit"` (and only one button should have that)), that would also provide an expected way to submit the form via the keyboard without necessarily needing to tab through to submit.

**Comment by willdurand** [comment]
> That's because `<Button>` defaults the type to `submit` unless you set a specific type.

ohhh right 👻

**Comment by kumar303** [comment]
This is ready for another look. The tab order matches the visual order now.

**Pull request merged**